### DISCUSSION TITLE
Integrate another server within pocketbase OnAfterBootstrap

### DISCUSSION THREAD

**Discussion by mlusetti** [pocketbase/pocketbase#4997] [Q&A] [answered]
Hi all,
  I'm already using PB as a framework with quite a good number of customization, all through event hooks.

Now I need to use integrate the final piece of the puzzle which is another server (a long running app) which needs to listen to a TCP port for incoming connections.

Do you feel this could work out ? Anyone having experience with something like this ?

Thanks

**Comment by thangngoc89** [comment]
You could make the 2 servers (pocketbase) and your other long running server talk to each other using either HTTP for gPRC. You can make call to any tcp port using Golang if you’re extending using Golang.

**Comment by mlusetti** [comment]
I did it (not using gRPC but the concept is the same) with Asynq which I use for "queue/job processing". I chose that way cause hibiken/asynq is designed to be distributed (I've 4 servers now).

I would like to have this one integrated with PB due to its nature, it has to run only if PB is running otherwise it has to be down.
So my idea to use the hook to fire it.

Looking for confirmations or concerns.Thanks for your opinion.

**Reply by thangngoc89** [reply]
Oh actually I’m doing the same but because I’m running AI inference so I use Celery job queue instead. I add to the queue inside event hooks of Pocketbase.

**Comment by mlusetti** [comment, marked as answer]
I ended up integrating it simply in my Go app without using pb hook system.
